### PR TITLE
Add form list container for common UI

### LINKED ISSUE #264: Need a new component for modal form, list management
Adding 1099s, W2s, spouse, and dependent all use a common UI, where the user is able to add N items, then edit the items, or remove the items. I'm about to add another such one for managing investment properties for Schedule E. So we need a UI element that encapsulates opening a form, showing a list of items, editing items, and removing items.

### PR DESCRIPTION
**Pull request by zakpatterson**
This factors out common edit / remove / add actions for list features. Same form list container is used for 1099, W2, spouse, and dependent.

Also adds edit functions for 1099 and W2 components.

Closes #264

### CODE REVIEW

File: src/components/income/W2JobInfo.tsx [hunk 1]
```diff
@@ -1,57 +1,13 @@
...
-        w2s.map((w2, i) =>
-          <W2ListItem key={i} remove={() => drop(i)} w2={w2} />
-        )
-      }
-    </List>
-  )
-}
+import { FormListContainer } from '../FormContainer'
+import { Work } from '@material-ui/icons'
+import { addW2, editW2 } from '../../redux/actions'
```
Comment (thegrims) [inline src/components/income/W2JobInfo.tsx:10]: Should have the ability to remove W2 as well. Looks like that ability was removed for W2s in this PR 
image

Comment (zakpatterson) [reply src/components/income/W2JobInfo.tsx]: Ok got that.

**Comment by zakpatterson** [comment]
Looks like editing 1099s is broken. I'll give that another pass now.

**Review by thegrims — APPROVED** [review]
LGTM

**Pull request merged**